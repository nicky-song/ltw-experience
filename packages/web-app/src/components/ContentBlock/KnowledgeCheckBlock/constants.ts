import {
  SELECT_ALL_LABEL,
  SELECT_ONE_LABEL,
} from '@learn-to-win/common/constants/multipleChoiceTypeText';
export const MIN_OPTIONS = 2;

export const multipleChoiceDropdownValues = [
  {
    title: SELECT_ONE_LABEL,
    value: 'selectone',
    label: SELECT_ONE_LABEL,
  },
  {
    title: SELECT_ALL_LABEL,
    value: 'selectall',
    label: SELECT_ALL_LABEL,
  },
];
