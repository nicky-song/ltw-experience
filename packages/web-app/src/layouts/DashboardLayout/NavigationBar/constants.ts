export const roles = [
  {
    key: '0',
    label: 'Super Admin',
    abbreviation: 'SA',
    pathprefix: 'superadmin',
    tabs: [
      {
        label: 'Organizations',
        key: '0',
        path: '/superadmin/organizations',
      },
    ],
  },
  {
    key: '1',
    label: 'Admin',
    abbreviation: 'A',
    pathprefix: 'admin',
    tabs: [
      {
        label: 'Dashboard',
        key: '0',
        path: '/admin/dashboard',
      },
      {
        label: 'Courses',
        key: '1',
        path: '/admin/courses',
      },
      {
        label: 'Users',
        key: '2',
        path: '/admin/users',
        // Per Khoa's designs, we might want this to be a dropdown in the future.
        // children:
        //     [
        //         {
        //             label: 'Item 1',
        //             key: '3'
        //         }
        //     ]
      },
    ],
  },
  {
    key: '2',
    label: 'Learner',
    abbreviation: 'L',
    pathprefix: 'learner',
    tabs: [
      {
        label: 'Dashboard',
        key: '0',
        path: '/learner/dashboard',
      },
      {
        label: 'Courses',
        key: '1',
        path: '/learner/courses',
      },
    ],
  },
];
