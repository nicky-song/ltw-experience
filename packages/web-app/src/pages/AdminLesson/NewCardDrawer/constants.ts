export const lessonCardsKey = 'lessonCards';
export const importCardsKey = 'importCards';
