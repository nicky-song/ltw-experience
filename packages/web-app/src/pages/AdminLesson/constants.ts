export const MAX_TITLE_LENGTH = 95;
export const TITLE_TOOL_TIP =
  'Grab attention with a concise card title! Max 95 characters.';
