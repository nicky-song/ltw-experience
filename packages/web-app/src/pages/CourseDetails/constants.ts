export const adminLearningItemCols = [
  {
    title: 'Name',
    dataIndex: 'name',
    key: 'name',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
  },
  {
    title: 'Number of Slides',
    dataIndex: 'numberOfSlides',
    key: 'numberOfSlides',
  },
];
