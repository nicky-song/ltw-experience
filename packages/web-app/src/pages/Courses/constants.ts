export const coursesColumns = [
  {
    title: 'Name',
    dataIndex: 'name',
    key: 'name',
    width: '25%',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    width: '50%',
  },
  {
    title: 'Details',
    dataIndex: 'details',
    key: 'details',
    width: '25%',
  },
];
