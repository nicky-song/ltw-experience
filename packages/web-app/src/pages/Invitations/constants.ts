export const enrollmentColumns = [
  {
    title: 'Name',
    dataIndex: 'name',
    key: 'name',
  },
  {
    title: 'Status',
    dataIndex: 'status',
    key: 'status',
  },
];
