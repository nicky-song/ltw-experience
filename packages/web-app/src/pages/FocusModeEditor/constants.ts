import { FeedBackType } from '@learn-to-win/common/constants/cardFeedbackType';

export const feedbackTooltipText = (feedBackType: FeedBackType) => {
  switch (feedBackType) {
    case 'correct':
      return 'Define feedback for the learner. This screen displays after answering the correct answer.';
    case 'incorrect':
      return 'Define feedback for the learner. This screen displays after answering the incorrect answer.';
    case 'confidence':
      return 'Gauge learner confidence level. This screen displays after selecting an answer';
    default:
      return '';
  }
};
