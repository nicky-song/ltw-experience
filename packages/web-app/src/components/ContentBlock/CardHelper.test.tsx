import { Descendant } from 'slate';
import {
  addContentBlockToCard,
  deleteContentBlockFromCard,
  getContentBlocksFromCard,
  updateContentBlockInCard,
} from './CardHelper';
import { Card } from '@learn-to-win/common/features/Cards/cardTypes';
import { CardType } from '@learn-to-win/common/constants';

const card: Card = {
  id: '123',
  title: 'Test Title',
  type: CardType.LESSON_CARD,
  learningItem: '123',
  sequenceOrder: 1,
  learningItemId: '123',
  confidenceCheck: true,
  json: {
    version: '1',
    description: 'Test Description',
    templateType: null,
    contentBlocks: [
      {
        id: '1',
        type: 'title',
        text: 'Test Title',
      },
      {
        id: '2',
        type: 'subtitle',
        text: 'Test Subtitle',
      },
      {
        id: '3',
        type: 'body',
        text: 'Test Body',
      },
    ],
  },
};

describe('getContentBlocksFromCard', () => {
  it('should return the correct content blocks', () => {
    const jsx = getContentBlocksFromCard(card, false);
    expect(jsx).toMatchSnapshot();
  });
});

describe('addContentBlockToCard', () => {
  const newCard = addContentBlockToCard(card, {
    id: '4',
    type: 'body',
    text: 'Test Body',
  });

  expect(newCard.json.contentBlocks.length).toBe(4);
});

describe('deleteContentBlockFromCard', () => {
  it('should have the correct number of content blocks', () => {
    const newCard = deleteContentBlockFromCard(card, '3');

    expect(newCard.json.contentBlocks.length).toBe(2);
  });
});

describe('updateContentBlock', () => {
  const newCard = updateContentBlockInCard(card, {
    id: '1',
    type: 'title',
    text: 'Updated Test Title',
    json: [{ children: [{ text: 'Updated Test Title' }] } as Descendant],
  });

  if ('text' in newCard.json.contentBlocks[0]) {
    expect(newCard.json.contentBlocks[0].text).toBe('Updated Test Title');
  }
});
