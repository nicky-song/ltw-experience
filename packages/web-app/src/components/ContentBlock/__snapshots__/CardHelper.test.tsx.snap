// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getContentBlocksFromCard should return the correct content blocks 1`] = `
[
  <TitleBlock
    card={
      {
        "confidenceCheck": true,
        "id": "123",
        "json": {
          "contentBlocks": [
            {
              "id": "1",
              "text": "Test Title",
              "type": "title",
            },
            {
              "id": "2",
              "text": "Test Subtitle",
              "type": "subtitle",
            },
            {
              "id": "3",
              "text": "Test Body",
              "type": "body",
            },
          ],
          "description": "Test Description",
          "templateType": null,
          "version": "1",
        },
        "learningItem": "123",
        "learningItemId": "123",
        "sequenceOrder": 1,
        "title": "Test Title",
        "type": "lesson",
      }
    }
    id="1"
    showEditor={false}
  />,
  <SubtitleBlock
    card={
      {
        "confidenceCheck": true,
        "id": "123",
        "json": {
          "contentBlocks": [
            {
              "id": "1",
              "text": "Test Title",
              "type": "title",
            },
            {
              "id": "2",
              "text": "Test Subtitle",
              "type": "subtitle",
            },
            {
              "id": "3",
              "text": "Test Body",
              "type": "body",
            },
          ],
          "description": "Test Description",
          "templateType": null,
          "version": "1",
        },
        "learningItem": "123",
        "learningItemId": "123",
        "sequenceOrder": 1,
        "title": "Test Title",
        "type": "lesson",
      }
    }
    id="2"
    showEditor={false}
  />,
  <BodyBlock
    card={
      {
        "confidenceCheck": true,
        "id": "123",
        "json": {
          "contentBlocks": [
            {
              "id": "1",
              "text": "Test Title",
              "type": "title",
            },
            {
              "id": "2",
              "text": "Test Subtitle",
              "type": "subtitle",
            },
            {
              "id": "3",
              "text": "Test Body",
              "type": "body",
            },
          ],
          "description": "Test Description",
          "templateType": null,
          "version": "1",
        },
        "learningItem": "123",
        "learningItemId": "123",
        "sequenceOrder": 1,
        "title": "Test Title",
        "type": "lesson",
      }
    }
    id="3"
    showEditor={false}
  />,
]
`;
